public/app/features/alerting/unified/utils/routeTree.ts

```diff
@@ -149,6 +163,70 @@ export function cleanRouteIDs(route: Route | RouteWithID): Route {
   );
 }
 
+// remove IDs from the Kubernetes routes
+export function cleanKubernetesRouteIDs(
+  routingTree: ComGithubGrafanaGrafanaPkgApisAlertingNotificationsV0Alpha1RoutingTree
+): ComGithubGrafanaGrafanaPkgApisAlertingNotificationsV0Alpha1RoutingTree {
+  return produce(routingTree, (draft) => {
+    draft.spec.routes = draft.spec.routes.map(cleanRouteIDs);
+  });
+}
+
 export function findExistingRoute(id: string, routeTree: RouteWithID): RouteWithID | undefined {
   return routeTree.id === id ? routeTree : routeTree.routes?.find((route) => findExistingRoute(id, route));
 }
+
+/**
+ * This function converts an object into a unique hash by sorting the keys and applying a simple integer hash
+ */
+export function hashRoute(route: Route): string {
+  const jsonString = JSON.stringify(stabilizeRoute(route));
+
+  // Simple hash function - convert to a number-based hash
+  let hash = 0;
+  for (let i = 0; i < jsonString.length; i++) {
+    const char = jsonString.charCodeAt(i);
+    hash = (hash << 5) - hash + char;
+    hash = hash & hash; // Convert to 32-bit integer
+  }
+
+  return hash.toString(36);
+}
+
+/**
+ * This function will sort the route's values and set the keys in a deterministic order
+ */
+export function stabilizeRoute(route: Route): Required<Route> {
```

Comment:
nit: I would err on the side of not exporting this and testing it implicitly, so there's never any accidental usage of this in real code other than for hashing - as I guess that sorting could have unintended side effects? (e.g. I don't know if mute timings care about the order)